rotkehlchen/tests/unit/test_cost_basis.py

```diff
@@ -533,15 +535,17 @@ def test_accounting_lifo_order(accountant):
     cost_basis.reset(DBSettings(cost_basis_method=CostBasisMethod.LIFO))
     asset_events = cost_basis.get_events(asset)
     # first we do a simple test that from 2 events the second is used
+    d = dt.datetime(2023, 4, 15, tzinfo=dt.timezone.utc)
```

Comment:
Why is this change needed? Why would timestamp 1, 2 etc. not work?

Reply:
Because it doesn't have a timezone. Here's the stacktrace:
```
rotkehlchen\tests\unit\test_cost_basis.py:575:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
rotkehlchen\accounting\cost_basis\base.py:218: in calculate_spend_cost_basis
    time=timestamp_to_date(acquisition_event.timestamp),
rotkehlchen\utils\mixins\customizable_date.py:24: in timestamp_to_date
    return timestamp_to_date(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

ts = 2, formatstr = '%d/%m/%Y %H:%M:%S %Z', treat_as_local = True

    def timestamp_to_date(
            ts: Timestamp,
            formatstr: str = '%d/%m/%Y %H:%M:%S',
            treat_as_local: bool = False,
    ) -> str:
        """Transforms a timestamp to a datestring depending on given formatstr and UTC/local choice"""
        if treat_as_local is False:
            date = datetime.datetime.fromtimestamp(ts, tz=datetime.timezone.utc).strftime(formatstr)
        else:  # localtime
            date = datetime.datetime.fromtimestamp(
                ts,  # ignore below is due to: https://github.com/pjknkda/flake8-datetimez/issues/11
>               tz=datetime.datetime.fromtimestamp(ts).astimezone().tzinfo,  # noqa: DTZ006
            ).strftime(formatstr)
E           OSError: [Errno 22] Invalid argument
```

This is the behaviour on windows:
```
>>> dt.datetime.fromtimestamp(1).astimezone()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 22] Invalid argument
```

Reply:
Hmm I am still not sure what the error is. 

So `dt.datetime.fromtimestamp(1)` works right? It's just that the `astimezone()` doesn't? And somehow by adding `d = dt.datetime(2023, 4, 15, tzinfo=dt.timezone.utc)` you initialize timezone to utc? What does `dt.datetime.fromtimestamp(1)` return?

Reply:
> So dt.datetime.fromtimestamp(1) works right? It's just that the astimezone() doesn't?

Yes

> And somehow by adding d = dt.datetime(2023, 4, 15, tzinfo=dt.timezone.utc) you initialize timezone to utc?

Yes

>What does dt.datetime.fromtimestamp(1) return?

`datetime.datetime(1970, 1, 1, 1, 0, 1)`

Reply:
Okay so astimezone() from a timestamp without a timezone returns an error. The docs do not mention any such thing. 
https://docs.python.org/3/library/datetime.html#datetime.datetime.astimezone

Without an arg it should take localtime which is what we want.

Only relevant issue I can find is this: https://github.com/python/cpython/issues/80940

So on the naive timezone object, the `astimezone` calls `time.localtime()` and perhaps calls it with a negative argument? Could be a timezone problem that the time becomes a pre-epoch time for you if your timezone is west of UTC?

Seems like the problem is deeper than just the test and the real fix would need to be done in the function, but need to first understand why/what's happening.

Reply:
>Could be a timezone problem that the time becomes a pre-epoch time for you if your timezone is west of UTC?
Not the case as I'm in UTC+2

It's very strange, it actually fails for all timestamps up until 86400
```
>>> dt.datetime.fromtimestamp(86400).astimezone()
datetime.datetime(1970, 1, 2, 1, 0, tzinfo=datetime.timezone(datetime.timedelta(seconds=3600), 'W. Europe Standard Time'))
>>> dt.datetime.fromtimestamp(86399).astimezone()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 22] Invalid argument
```

Reply:
It seems to be then exatctly like this bug: https://github.com/python/cpython/issues/73283
But in a different place. I don't see this bug reported so perhaps it would need to be reported. Is this in python 3.10 and 3.11? I guess then we found a bug in python in Windows and would need to report it.

Reply:
3.9, 3.10 and 3.11 it fails for me on windows 11

Reply:
Okay so then 2 questions.

1. Should I report a bug in python? Or do you wanna do it? Pointing to the previous bug and explaining that we hit the same thing with astimezone()

2. Production-wise ... what should be done here? Adding two extra ifchecks in all the `timestamp_to_date` calls seems silly. Perhaps just adjust test to use timestamp 86400 and up? And then add a comment pointing to the bug and leaving the handling for the future? Rotki in its current form won't handle any such timestamps so close to the EPOCH. What do you think?

Reply:
> Should I report a bug in python? Or do you wanna do it? Pointing to the previous bug and explaining that we hit the same thing with astimezone()

I'm happy to let you claim the glory of finding a bug in python if you want ;) otherwise I can do it if you want to focus on other things.

> Perhaps just adjust test to use timestamp 86400 and up?

I think that makes the makes the most sense. If this is a bonafide bug in python then it will eventually get fixed upstream, and like you said no one is going to be using any dates that close to EPOCH, so I don't think there's any danger to production here.

Reply:
> happy to let you claim the glory of finding a bug in python if you want ;)

Made an issue here: https://github.com/python/cpython/issues/107078

> I think that makes the makes the most sense. If this is a bonafide bug in python then it will eventually get fixed upstream, and like you said no one is going to be using any dates that close to EPOCH, so I don't think there's any danger to production here.

All right so when you make any change please do that then in both places you added the timestamp fix.